addons/hr_expense/tests/test_ui.py

```diff
@@ -60,3 +60,18 @@ def test_show_expense_receipt_on_expense_line_click(self):
         })
 
         self.start_tour('/web', 'show_expense_receipt_tour', login=self.env.user.login)
+
+    def test_expense_manager_can_always_set_employee(self):
+        """Test that users with access rights to `hr.expense` can set the employee on them
+        by using the usual form view, even if they do not have access rights to `hr.employee`
+        """
+        employee_1 = self.expense_employee
+        employee_2 = self.env['hr.employee'].create({'name': 'employee2'})
+        expense = self.env['hr.expense'].create({
+            'name': 'expense_for_tour_0',
+            'employee_id': employee_2.id,
+            'product_id': self.product_a.id,
+            'total_amount': 1,
+        })
+        self.start_tour('/web', 'create_expense_no_employee_access_tour', login=self.expense_user_manager.login)
```

Comment:
Is a tour really necessary?
Maybe the Form emulator would have been enough and faster for such a little feature? 🤔
(Just wondering)

Reply:
We're using a tour because the problem comes from the domain search on `employee_id` in the form which implies reading and may cause access problems, but even without access it is always possible to set the `employee_id` so doing something like `expense_form.employee_id = 42` wouldn't really test what we want.
Another testing solution we thought about was to call `.search` directly, but here the problem is that frontend determines whether the model under consideration is `hr.employee` or `hr.employee.public` and so it's still not testing *exactly* what we want.